insights/core/plugins.py

```diff
@@ -99,8 +100,9 @@ def invoke(self, broker):
         # Grab the timeout from the decorator, or use the default of 120.
         self.timeout = getattr(self, "timeout", 120)
 
-        signal.signal(signal.SIGALRM, self._handle_timeout)
-        signal.alarm(self.timeout)
+        if HostContext in broker:
```

Comment:
@lhuett - do you think it's reasonable to also move the definition of the `self.timeout` (L101) into the `if` clause? since it's only used in this case.

Reply:
@xiangce Good point, I can do that.